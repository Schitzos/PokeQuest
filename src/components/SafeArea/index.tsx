import theme from '@/theme';
import React from 'react';
import {StyleSheet, SafeAreaView, ViewStyle} from 'react-native';

interface SafeAreaProps {
  children: React.ReactNode;
  color?: string;
}

export default function SafeArea({children, color}: SafeAreaProps) {
  const styles = StyleSheet.create({
    base: {
      backgroundColor: color || theme.colors.neutral50,
    },
  });

  return (
    <SafeAreaView style={styles.base as ViewStyle}>{children}</SafeAreaView>
  );
}
