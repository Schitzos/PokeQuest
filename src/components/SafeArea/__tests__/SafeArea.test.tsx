import React from 'react';
import {render} from '@testing-library/react-native';
import SafeArea from '../index';
import '@testing-library/jest-dom';

describe('SafeArea component', () => {
  test('matches snapshot with default props', () => {
    const {toJSON} = render(<SafeArea>Child Component</SafeArea>);
    expect(toJSON()).toMatchSnapshot();
  });

  test('matches snapshot with custom color', () => {
    const {toJSON} = render(<SafeArea color="red">Child Component</SafeArea>);
    expect(toJSON()).toMatchSnapshot();
  });
});
