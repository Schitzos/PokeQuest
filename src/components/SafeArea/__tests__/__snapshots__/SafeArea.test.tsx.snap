// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SafeArea component matches snapshot with custom color 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "red",
    }
  }
>
  Child Component
</RCTSafeAreaView>
`;

exports[`SafeArea component matches snapshot with default props 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#F6F6F6",
    }
  }
>
  Child Component
</RCTSafeAreaView>
`;
